### PR TITLE
	feat: support latest and asset type filter

### PR DESCRIPTION
**Pull request by remy**
Adds library search API endpoint support for:

- `latest=true` - filters the assets to the latest assets only
- `assetFilter=[js|css|images|fonts]` (can be comma separated) - only include assets matching the filter type (to help exclude assets like images from material-design-icons libraries, etc)

Note that I've tried to keep the code in ES5 flavour (wasn't sure if that was right or not).

Reasoning: I'm doing a big rewrite of jsbin, and I've included cdnjs integration, but searching for libraries like "react" doesn't give me "react-dom" unless I request the assets, which doesn't allow for finer filtering (hence this PR).

### CODE REVIEW

File: filter-assets.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var _ = require('lodash');
+
+var filterTypes = {
+  js: 'js',
+  css: 'css',
+  fonts: 'ttf', // ??
+  images: ['png', 'jpeg', 'jpg', 'gif']
```
Comment (remy) [inline filter-assets.js]: Pretty sure this is non exhaustive…but I guess the main point is to filter these _out_ in favour of only getting CSS or JS files. Probably still worth getting it complete though…is that anywhere?

**Comment by tholman** [comment]
+1

**Comment by remy** [comment]
Anyone look for this, but can't wait (or it doesn't get merged), you can do this using [jq](https://stedolan.github.io/jq/) and curl:

```
curl -s https://cdnjs.com/packages.min.json | jq '.packages | .[] | .version as $version | { name: .name, version: .version, assets: .assets | .[] | select(.version == $version) | (.files | map(select( . | endswith(".js") )) ) }'
```

Broken up for legibility:

```bash
curl ... | jq '
.packages | # select the packages
.[] | # forEach
.version as $version | # store the latest version as $version
{ name: .name, version: .version, assets: .assets | # pick some fields, and generate assets from…
  .[] | # forEach asset item
  select(.version == $version) | # select where version === latest version
    (.files | # pick the files…
      map(select( . | endswith(".js"))) # filter for only the .js files
 ) }'
```

There's probably a better way of the doing the `jq` select, but it's my first time using `jq` in anger. Certainly gives me the result I needed.

**Comment by PeterDaveHello** [comment]
Oh wow ... sorry for my late response ... there are too many notifications in my box 😭

File: apiServer.js [hunk 2]
```diff
@@ -8,6 +8,9 @@ var args = process.argv.slice(2);
 var localMode = false;
 var compress = require('compression');
 var bodyParser = require('body-parser');
+
```
Comment (PeterDaveHello) [inline apiServer.js]: Can we remove this empty line please?

File: filter-assets.js (hunk 1 above)
Comment (PeterDaveHello) [reply filter-assets.js]: `cur`, `ico`, `svg`, `webp`

File: filter-assets.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var _ = require('lodash');
+
+var filterTypes = {
+  js: 'js',
+  css: 'css',
+  fonts: 'ttf', // ??
```
Comment (PeterDaveHello) [inline filter-assets.js]: `woff`, `woff2`, `eot`, `otf`

File: filter-assets.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+var _ = require('lodash');
+
+var filterTypes = {
```
Comment (PeterDaveHello) [inline filter-assets.js]: Need audio(`mp3`, `wav`,`ogg`, `aac`), flash(`swf`), map(`map`) type.

File: package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "startWeb": "node webServer.js",
-    "startAPI": "node apiServer.js",
+    "start": "node apiServer.js",
```
Comment (PeterDaveHello) [inline package.json]: Why do we change this here?

**Comment by remy** [comment]
Hey, just checking in - I was away for work then birthday, then just settling back in. I'll fix these review points (ideally this week). Thanks.

**Comment by PeterDaveHello** [comment]
@remy thanks a lot!

**Comment by PeterDaveHello** [comment]
hmmm, anything I can help here?

**Comment by drewfreyling** [comment]
@remy is this still being worked on?

**Comment by remy** [comment]
Ah, no. I remember hitting some weird issue that was deep inside the
system, and possibly lost the thread of work I was doing at the time.

**Comment by MattIPv4** [comment]
Hey @remy, if you're interested in picking this up again, we have a new API server now, which can be found in https://github.com/cdnjs/api-server :)

As this API server here is no longer used, I'll close this PR.

**Pull request closed without merging**